File: 프로그래머스/2/12953. N개의 최소공배수/N개의 최소공배수.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(vector<int> arr) {
    int answer = 0;
    sort(arr.begin(), arr.end());
    int max = arr.back();
    
    while(true)
    {
        bool flag = true;
        for(int i = 0; i < arr.size() && flag; ++i)
        {
            if (max % arr[i] != 0)
                flag = false;
        }
        
        if (flag)
            break;
        
        max += arr.back();
    }
    
    return max;
}

File: 프로그래머스/2/12953. N개의 최소공배수/N개의 최소공배수.java
class Solution {
    public int solution(int[] arr) {
        for (int i = arr.length - 1; i > 0; i--) {
            arr[i - 1] = (arr[i] * arr[i - 1] ) / gcb(arr[i], arr[i - 1]);
        }
        return arr[0];
    }
    
    public int gcb(int a, int b) {
        
        while (b > 0) {
            int tmp = a % b;
            a = b;
            b = tmp;
        }
        return a;
    }
}